editoast/src/views/path/path_item_cache.rs

```diff
@@ -408,3 +461,136 @@ fn secondary_code_filter(
+        let db_pool = DbConnectionPoolV2::for_tests();
+        let mut conn = db_pool.get_ok();
+        let infra = create_empty_infra(&mut conn).await;
+
+        // Create three operational points with different identifier combinations
+        let op1 = create_op("op_1", Some("ABC"), Some(1234));
+        let op2 = create_op("op_2", Some("DEF"), None); // No UIC
+        let op3 = create_op("op_3", None, Some(5678)); // No trigram
+
+        // Insert OPs into the database
+        create_infra_object(&mut conn, infra.id, op1).await;
+        create_infra_object(&mut conn, infra.id, op2).await;
+        create_infra_object(&mut conn, infra.id, op3).await;
+
+        // Create path items that reference these OPs by different methods
+        let path_items = [
+            PathItemLocation::OperationalPointReference(OperationalPointReference {
+                reference: OperationalPointIdentifier::OperationalPointId {
+                    operational_point: Identifier::from("op_1"),
+                },
+                track_reference: None,
+            }),
+            PathItemLocation::OperationalPointReference(OperationalPointReference {
+                reference: OperationalPointIdentifier::OperationalPointDescription {
+                    trigram: NonBlankString::from("DEF"),
+                    secondary_code: None,
+                },
+                track_reference: None,
+            }),
+            PathItemLocation::OperationalPointReference(OperationalPointReference {
+                reference: OperationalPointIdentifier::OperationalPointUic {
+                    uic: 5678,
+                    secondary_code: None,
+                },
+                track_reference: None,
+            }),
+        ];
+
+        let path_item_refs: Vec<&PathItemLocation> = path_items.iter().collect();
+
+        // Load the cache using the real load() method
+        let cache = PathItemCache::load(&mut conn, infra.id, &path_item_refs)
+            .await
+            .expect("Failed to load cache");
+
+        // Test OP1 (has both trigram and UIC) can be found by all methods
+        assert!(
+            cache.get_from_id("op_1").is_some(),
+            "OP1 should be findable by ID"
+        );
+        assert!(
+            cache.get_from_trigram("ABC").is_some(),
+            "OP1 should be findable by trigram even though queried by ID"
+        );
+        assert!(
+            cache.get_from_uic(1234).is_some(),
+            "OP1 should be findable by UIC even though queried by ID"
+        );
+
+        // Verify it's the same OP
+        let op_by_id = cache.get_from_id("op_1").unwrap();
+        let op_by_trigram = cache.get_from_trigram("ABC").unwrap().first().unwrap();
+        let op_by_uic = cache.get_from_uic(1234).unwrap().first().unwrap();
+        assert_eq!(op_by_id.obj_id, "op_1");
+        assert_eq!(op_by_trigram.obj_id, "op_1");
+        assert_eq!(op_by_uic.obj_id, "op_1");
```

Comment:
I believe you can inline and avoid interleaving lines.

```suggestion
        assert!(cache.get_from_id("op_1").unwrap().obj_id, "op_1");
        assert!(cache.get_from_trigram("ABC").unwrap().first().unwrap().obj_id, "op_1");
        assert!(cache.get_from_uic(1234).unwrap().first().unwrap().obj_id, "op_1");
```

Reply:
Done in [8b1448d6](https://github.com/OpenRailAssociation/osrd/compare/8b1448d64c27731e6b0137d1a158bad48118a931..d26c8fa7b335a3d97781ee3edb1beb73fbe52666)